MBBSEmu/CPU/CPUCore.cs

```diff
@@ -3365,6 +3420,63 @@ private void Op_Fld1()
             FpuStack[Registers.Fpu.GetStackTop()] = 1d;
         }
 
+        /// <summary>
+        ///     Floating Point Load Operation (x87)
+        ///
+        ///     Exchanges the contents of registers ST(0) and ST(i). If no source operand is specified, the contents of ST(0) and ST(1) are exchanged.
+        /// </summary>
+        [MethodImpl(OpcodeCompilerOptimizations)]
+        private void Op_Fxch()
+        {
+            var source = GetOperandValueDouble(_currentInstruction.Op0Kind, EnumOperandType.Source);
+            var ST0 = FpuStack[Registers.Fpu.GetStackTop()];
+            var ST1 = FpuStack[Registers.Fpu.GetStackPointer(Register.ST1)];
+
+            //TODO Handle invalid and infinity cases
+
+            (ST0, ST1) = (ST1, ST0);
```

Comment:
doesn't this just change the local variable values and not the values actually in the FpuStack?